### PR TITLE
JS API/Core/Kits: New - Closing the menu redirect the user to the widget/control they were previously editing

### PR DESCRIPTION
**Pull request by iNewLegend**
Causes issue with **GlobalControlSelect::initGlobalPopover**: after back-to-route.
```
elementorCommon.dialogsManager.createWidget( ... ).elements ( not connected ).
```

@matipojo, usage:
```
		const args = {
			route: $e.routes.getHistory( 'panel' ).reverse()[ 0 ].route,
			container: this.view.options.container,
		};
		$e.run( 'panel/global/open', args ).then( () => $e.route( 'panel/global/colors-and-typography' ) );
```

### CODE REVIEW

File: assets/dev/js/editor/container/container.js [hunk 1]
```diff
@@ -276,11 +276,11 @@ export default class Container extends ArgsObject {
 			return this;
 		}
 
-		if ( this !== this.renderer && this.renderer.view.isDestroyed ) {
+		if ( this !== this.renderer && this.renderer.view?.isDisconnected && this.renderer.view.isDisconnected() ) {
```
Comment (matipojo) [inline assets/dev/js/editor/container/container.js:279]: `this !== this.renderer && this.renderer.view.isDisconnected()`?

File: assets/dev/js/editor/utils/helpers.js [hunk 2]
```diff
@@ -539,7 +539,7 @@ module.exports = {
 
 			const scrolling = elementTop - ( parentHeight / 2 );
 
-			$scrolled.stop( true ).animate( { scrollTop: scrolling }, 1000 );
+			$scrolled.stop( true ).animate( { scrollTop: scrolling }, 0 === timeout ? 0 : 1000 );
```
Comment (matipojo) [inline assets/dev/js/editor/utils/helpers.js:542]:
```suggestion
			$scrolled.stop( true ).animate( { scrollTop: scrolling }, timeout );
```

File: core/kits/assets/js/hooks/ui/panel/global/open/save-route-history.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+	getConditions( args = {}, result ) {
+		return args.route && args.container;
+	}
+
+	apply( args ) {
+		// TODO: Save reference in memory.
+		// $e.data.setTemp( $e.components.get( 'panel/global' ),  'routeHistory', args );
+		// or $e.data.setTempOnce - means when you get it, it will automatically remove.
+		$e.routes.temp = args;
```
Comment (matipojo) [inline core/kits/assets/js/hooks/ui/panel/global/open/save-route-history.js:18]: Rename `temp`.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/container/container.js (hunk 1 above)
Comment (iNewLegend) [reply assets/dev/js/editor/container/container.js:279]: Not enough... causes error.

File: assets/dev/js/editor/utils/helpers.js (hunk 2 above)
Comment (iNewLegend) [reply assets/dev/js/editor/utils/helpers.js]: Handled: https://github.com/elementor/elementor/pull/11791/commits/24621380bfdbc3bc861c7e2b4b418608bbda3907

File: core/kits/assets/js/hooks/ui/panel/global/open/save-route-history.js (hunk 3 above)
Comment (iNewLegend) [reply core/kits/assets/js/hooks/ui/panel/global/open/save-route-history.js]: Handled: https://github.com/elementor/elementor/pull/11791/commits/2762229f0c220d677e906784020062ac37e31f2b

**Review by matipojo — DISMISSED** [review]

File: assets/dev/js/editor/views/base-container.js [hunk 4]
```diff
@@ -123,13 +123,17 @@ module.exports = Marionette.CompositeView.extend( {
...
-		if ( element.isDestroyed ) {
+		if ( element.isDisconnected() ) {
 			element = $e.components.get( 'document' ).utils.findViewById( element.model.id );
 		}
 
 		return element;
 	},
 
+	isDisconnected: function() {
+		return this.isDestroyed || ! this.el.isConnected;
```
Comment (kobizz) [inline assets/dev/js/editor/views/base-container.js:134]: Where is the `isConnected` definition?

File: assets/dev/js/editor/regions/panel/pages/editor/commands/open.js [hunk 5]
```diff
@@ -2,6 +2,8 @@ import CommandBase from 'elementor-api/modules/command-base';
 
 export class Open extends CommandBase {
 	apply( args ) {
+		elementor.helpers.scrollToView( args.view.$el, 0 );
```
Comment (kobizz) [inline assets/dev/js/editor/regions/panel/pages/editor/commands/open.js:5]: Does `onEditRequest` no longer run?

File: core/kits/assets/js/globals/global-select-behavior.js [hunk 6]
```diff
@@ -17,9 +17,20 @@ export default class GlobalControlSelect extends Marionette.Behavior {
 	// This method exists because the UI elements are printed after controls are already rendered.
 	registerEvents() {
 		this.ui.globalPreviewsContainer.on( 'click', '.e-global__preview-item', ( event ) => this.applySavedGlobalValue( event.currentTarget.dataset.globalId ) );
-		this.ui.globalControlSelect.on( 'click', ( event ) => this.toggleSelect( event ) );
+
+		// TODO: Find better solution.
+		if ( this.ui.globalControlSelect ) {
```
Comment (kobizz) [inline core/kits/assets/js/globals/global-select-behavior.js:22]: Why does `globalControlSelect` not always exist?

**Review by kobizz — CHANGES_REQUESTED** [review]

File: core/kits/assets/js/hooks/ui/panel/global/close/back-to-route-history.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+		 * Since cache deleted after leaving globals.
+		 * Cover issue: When back to route, it back to style, it causes the UI ask for styles separately and since,
+		 * Cache deleted, it asks the remote ( $e.data ) for specific colors/typography endpoints and causes a delay in global select box.
+		 * To handle the the issue, request globals manually, then back to route.
+		 */
+		$e.data.get( 'globals/index' ).then( () => {
+			// Since the container comes from history, its not connected element.
+			historyBeforeOpen.container = historyBeforeOpen.container.lookup();
+
+			$e.run( 'panel/editor/open', {
```
Comment (iNewLegend) [inline core/kits/assets/js/hooks/ui/panel/global/close/back-to-route-history.js:32]: trigger request:edit instead.

File: assets/dev/js/editor/views/base-container.js (hunk 4 above)
Comment (iNewLegend) [reply assets/dev/js/editor/views/base-container.js:134]: It's Native of element.

File: core/kits/assets/js/globals/global-select-behavior.js (hunk 6 above)
Comment (iNewLegend) [reply core/kits/assets/js/globals/global-select-behavior.js:22]: I don't know its better to ask Udi i think.

File: assets/dev/js/editor/regions/panel/pages/editor/commands/open.js (hunk 5 above)
Comment (iNewLegend) [reply assets/dev/js/editor/regions/panel/pages/editor/commands/open.js]: Handled: https://github.com/elementor/elementor/pull/11791/commits/f0f8f4bc6503fe014bd1995adf19347a8aefe221

**Review by matipojo — DISMISSED** [review]

File: core/kits/assets/js/globals/global-select-behavior.js (hunk 6 above)
Comment (kobizz) [reply core/kits/assets/js/globals/global-select-behavior.js:22]: We can't ignore the problem. We have to fix it

Comment (matipojo) [reply core/kits/assets/js/globals/global-select-behavior.js:22]: @udidol can you help?

Comment (udidol) [reply core/kits/assets/js/globals/global-select-behavior.js:22]: This is a problem we know about for a while now. 
I have tried to debug it and find why it happened and couldn't - I have no idea what causes this bug.
I agree we absolutely have to fix it.

Comment (kobizz) [reply core/kits/assets/js/globals/global-select-behavior.js:22]: Remove this fix from this PR. We will try to resolve it on our side

Comment (iNewLegend) [reply core/kits/assets/js/globals/global-select-behavior.js]: Handled: https://github.com/elementor/elementor/pull/11791/commits/9bdc1e71fd9955ff123ec9b4a9bf24bbaf13f05b

**Review by kobizz — APPROVED** [review]

**Review by matipojo — APPROVED** [review]

**Pull request merged**